src/test/java/com/library/controller/BookControllerTest.java

```diff
@@ -0,0 +1,208 @@
+import org.springframework.test.web.servlet.MvcResult;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+class BookControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private BookMapper bookMapper;
+
+    @MockBean
+    private BookRepository bookRepository;
+
+    private TestStorage testStorage;
+
+    private static final String MISSING_TYPE = "Not entered data: [authors]";
+
+    private static final String DELETED_SUCCESS = "Book is deleted successfully!";
+
+    @BeforeEach
+    public void init() {
+        testStorage = new TestStorage();
+    }
+
+    @SneakyThrows
+    @Test
+    void getAllBooks() {
+        Pageable paging = PageRequest.of(0, 5);
+        when(bookRepository.findAll(paging)).thenReturn(new PageImpl<Book>(testStorage.getBooks()));
+
+        MockHttpServletResponse response = mockMvc.perform(get("/books")
+                        .param("page", "1")
+                        .param("sizeOnPage", "5"))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andReturn()
+                .getResponse();
+        JSONArray jsonArray = new JSONArray(response.getContentAsString());
+        JSONObject jsonObject = jsonArray.getJSONObject(0);
+
+        assertNotNull(response.getContentAsString());
+        assertEquals(1, jsonObject.get("id"));
+        assertEquals("War and peace", jsonObject.get("title"));
+        assertEquals(3, jsonArray.length());
+        assertEquals("application/json", response.getContentType());
+    }
+
+    @SneakyThrows
+    @Test
+    void getBookById() {
+        when(bookRepository.findById(1L)).thenReturn(Optional.of(testStorage.getBook()));
+
+        MockHttpServletResponse response = mockMvc.perform(get("/books/{id}", "1"))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andReturn()
+                .getResponse();
+        JSONObject jsonObject = new JSONObject(response.getContentAsString());
+
+        assertNotNull(jsonObject);
+        assertEquals("application/json", response.getContentType());
```

Comment:
Is this checking is needed?

Reply:
I saw it in example... maybe yes, maybe no)